src/js_object_accessor.hpp

```diff
@@ -421,33 +434,33 @@ struct Unbox<JSEngine, Timestamp> {
 
 template<typename JSEngine>
 struct Unbox<JSEngine, Obj> {
-    static Obj call(NativeAccessor<JSEngine> *ctx, typename JSEngine::Value const& value, realm::CreatePolicy policy, ObjKey current_row) {
+    static Obj call(NativeAccessor<JSEngine> *native_accessor, typename JSEngine::Value const& value, realm::CreatePolicy policy, ObjKey current_row) {
         using Value = js::Value<JSEngine>;
         using ValueType = typename JSEngine::Value;
 
-        auto object = Value::validated_to_object(ctx->m_ctx, value);
-        if (js::Object<JSEngine>::template is_instance<RealmObjectClass<JSEngine>>(ctx->m_ctx, object)) {
-            auto realm_object = get_internal<JSEngine, RealmObjectClass<JSEngine>>(ctx->m_ctx, object);
-            if (realm_object->realm() == ctx->m_realm) {
-                return realm_object->obj();
-            }
+        RealmLink<JSEngine> realm_link {native_accessor->m_ctx, value};
+        auto current_realm = native_accessor->m_realm;
 
-            bool updating = policy.copy && policy.update;
-            if (!updating && !policy.create) {
-                throw std::runtime_error("Realm object is from another Realm");
-            }
+        if(realm_link.belongs_to_realm(current_realm)){
+            return realm_link.get_realm_object();
+        }
+
+        if(realm_link.is_instance() && realm_link.is_read_only(policy)){
+            throw std::runtime_error("Realm object is from another Realm");
         }
 
         if (!policy.create) {
             return Obj();
         }
 
-        if (Value::is_array(ctx->m_ctx, object)) {
-            object = Schema<JSEngine>::dict_for_property_array(ctx->m_ctx, *ctx->m_object_schema, object);
+        auto object = Value::validated_to_object(native_accessor->m_ctx, value);
+        if (Value::is_array(native_accessor->m_ctx, object)) {
+            object = Schema<JSEngine>::dict_for_property_array(native_accessor->m_ctx, *native_accessor->m_object_schema, object);
         }
 
-        auto child = realm::Object::create<ValueType>(*ctx, ctx->m_realm, *ctx->m_object_schema,
-                                                      static_cast<ValueType>(object), policy, current_row);
+        auto child = realm::Object::create<ValueType>
+            (*native_accessor, native_accessor->m_realm, *native_accessor->m_object_schema,
+                    static_cast<ValueType>(object), policy, current_row);
```

Comment:
This formatting looks off/inconsistent with other files - does realm-js not have clang or something similar to enforce consistent code style?

Reply:
We don't have that yet and until we have, I suggest not focussing too much on code style when reviewing PRs to the Realm JS C++ code. We will the formatter on the entire codebase once we get there.